docs/csharp/roslyn-sdk/get-started/syntax-analysis.md

```diff
@@ -0,0 +1,222 @@
+
+The four primary building blocks of syntax trees are:
+
+* The <xref:Microsoft.CodeAnalysis.SyntaxTree?displayProperty=nameWithType> class, an instance of which represents an entire parse tree. <xref:Microsoft.CodeAnalysis.SyntaxTree> is an abstract class that has language-specific derivatives. You use the parse methods of the <xref:Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree?displayProperty=nameWithType> (or <xref:Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxTree?displayProperty=nameWithType>) class to parse text in C# or VB.
+* The <xref:Microsoft.CodeAnalysis.SyntaxNode?displayProperty=nameWithType> class, instances of which represent syntactic constructs such as declarations, statements, clauses, and expressions.
+* The <xref:Microsoft.CodeAnalysis.SyntaxToken?displayProperty=nameWithType> structure, which represents an individual keyword, identifier, operator, or punctuation.
+* And lastly the <xref:Microsoft.CodeAnalysis.SyntaxTrivia?displayProperty=nameWithType> structure, which represents syntactically insignificant bits of information such as the whitespace between tokens, preprocessing directives, and comments.
+
+Trivia, tokens, and nodes are composed hierarchically to form a tree that completely represents everything in a fragment of Visual Basic or C# code. You can see this structure using the **Syntax Visualizer** window. In Visual Studio, choose **View** > **Other Windows** > **Syntax Visualizer**. For example, the preceding C# source file examined using the **Syntax Visualizer** looks like the following figure:
+
+**SyntaxNode**: Blue | **SyntaxToken**: Green | **SyntaxTrivia**: Red
+![C# Code File](media/walkthrough-csharp-syntax-figure1.png)
+
+By navigating this tree structure, you can find any statement, expression, token, or bit of whitespace in a code file.
+
+While you can find anything in a code file using the Syntax APIs, most scenarios involve examining small snippets of code, or searching for particular statements or fragments. The two examples that follow show typical uses to browse the structure of code, or search for single statements.
+
+## Traversing trees
+
+You can examine the nodes in a syntax tree in two ways. You can traverse the tree to examine each node, or you can query for specific elements or nodes.
+
+> [!IMPORTANT]
+> The following samples require the **.NET Compiler SDK** installed as part of Visual Studio 2017. You can find the .NET Compiler SDK as the last optional component listed under the **Visual Studio extension development** workload. The templates aren't installed without this component.
```

Comment:
this can be easily missed. Should we add a prerequisites section on top? I was wondering why I was not seeing the template and then went digging for the info in the middle of the doc.

Reply:
Name also seems to be .NET Compiler Platform SDK, no?

Reply:
dotnet/roslyn-api-docs#36 will add a topic to install this template. I'll make sure to reference that new article from all of these when it's written (probably with an include for prereqs.)